pkg/testfixtures/storage/postgres.go

```diff
@@ -98,7 +99,13 @@ func (p *postgresTestContainer) RunPostgresTestContainer(t testing.TB) (Datastor
 	cont, err := dockerClient.ContainerCreate(context.Background(), &containerCfg, &hostCfg, nil, nil, name)
 	require.NoError(t, err, "failed to create postgres docker container")
 
+	var db *sql.DB
+
 	stopContainer := func() {
+		if db != nil {
+			db.Close()
+		}
```

Comment:
why do we need this? by the time this function runs, `db` is nil always

Reply:
actually, if you move `var db *sql.DB` to 139, you don't need this

Reply:
True, because we `defer db.Close()` above any points of potential error below it. It felt like an added safeguard as it guarantees if even if miss a point herein the caller calling `stop` will close it. That's why I had it.